pkg/secrets/handlers/secrets_handler_k8s.go

```diff
@@ -0,0 +1,130 @@
+package handlers
+
+import (
+	"fmt"
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/access_token"
+	secretsConfig "github.com/cyberark/conjur-authn-k8s-client/pkg/secrets/config"
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/secrets/conjur"
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/secrets/k8s"
+	log "github.com/cyberark/conjur-authn-k8s-client/pkg/sidecar/logging"
+	"os"
+	"strings"
+)
+
+type SecretsHandlerK8sUseCase struct {
+	AccessTokenHandler   access_token.AccessTokenHandler
+	ConjurSecretsFetcher conjur.ConjurSecretsFetcherInterface
+	K8sSecretsHandler    k8s.K8sSecretsHandler
+}
+
+var errLogger = log.ErrorLogger
+
+func NewSecretHandlerK8sUseCase(secretsConfig secretsConfig.Config, AccessTokenHandler access_token.AccessTokenHandler) (SecretsHandler *SecretsHandlerK8sUseCase, err error) {
+	k8sSecretsHandler, err := k8s.New(secretsConfig)
+	if err != nil {
+		errLogger.Printf(err.Error())
+		os.Exit(1)
+	}
+
+	var conjurSecretsFetcher conjur.ConjurSecretsFetcher
+
+	return &SecretsHandlerK8sUseCase{
+		AccessTokenHandler:   AccessTokenHandler,
+		ConjurSecretsFetcher: conjurSecretsFetcher,
+		K8sSecretsHandler:    *k8sSecretsHandler,
+	}, nil
+}
+
+func (secretsHandlerK8sUseCase *SecretsHandlerK8sUseCase) HandleSecrets() error {
+	k8sSecretsMap, err := secretsHandlerK8sUseCase.K8sSecretsHandler.RetrieveK8sSecrets()
+	if err != nil {
+		errLogger.Printf("Failure retrieving k8s secretsHandlerK8sUseCase: %s", err.Error())
+		return err
+	}
+
+	access_token, err := secretsHandlerK8sUseCase.AccessTokenHandler.Read()
+	if err != nil {
+		errLogger.Printf("Failure retrieving access token: %s", err.Error())
+		return err
+	}
+
+	variableIDs, err := getVariableIDsToRetrieve(k8sSecretsMap.PathMap)
+	if err != nil {
+		return fmt.Errorf("Error parsing Conjur variable ids: %s", err)
+	}
+
+	retrievedConjurSecrets, err := retrieveRequiredConjurSecrets(
```

Comment:
1. Why this function is not a part of the ConjurSecretsFetcher, i expected to syntax like: `retrievedConjurSecrets, err := secretsHandlerK8sUseCase.ConjurSecretsFetcher.retrieveRequiredConjurSecrets(access_token, variableIDs)`
2. Do you clear access_token & ConjurSecretsFetcher after the use?

Reply:
1. now that i look at the implementation of the method it's quite redundant. probably had more meat in it and got thinner. I'll remove it.
2. should i clear `access_token`? it is cleared in `main`. what do you mean by clearing `ConjurSecretsFetcher`?

Reply:
2. edited the comment, about access_token i think when we called read() we returned a copy of the data, if this is the case we should decide if to change the interface to return an pointer to data, or it will be the responsibility of the called to clear the data

Reply:
ignore 2 it got clean i will add UT that test it, just test the same issue regarding retrievedConjurSecrets